lib/mixin.js

```diff
@@ -0,0 +1,44 @@
+var Element = require('./Element')
+var mapValues = require('./utils/mapValues')
+
+var mixin = {
+  componentWillMount: function () {
+    this.connectedFauxDOM = {}
+    this.animateFauxDOMUntil = 0
+  },
+  connectFauxDOM: function (node, name) {
+    this.connectedFauxDOM[name] = typeof node !== 'string' ? node : new Element(node)
+    setTimeout(this.drawFauxDOM)
+    return this.connectedFauxDOM[name]
+  },
+  drawFauxDOM: function () {
+    var virtualDOM = mapValues(this.connectedFauxDOM, function (n) {
+      return n.toReact()
+    })
+    this.setState(virtualDOM)
+  },
+  animateFauxDOM: function (duration) {
+    this.animateFauxDOMUntil = Math.max(Date.now() + duration, this.animateFauxDOMUntil)
```

Comment:
The architecture only allows one animation at a time if I'm not mistaken with a locked framerate. Will that make some D3 animations fall over?

I really like the mixin approach, it's a great way to plug stuff in and integrate the faux DOM with a component, but I'm not sure about 16ms intervals with hard coded lengths and only one animation at a time.